admin/app/components/solidus_admin/base_component.rb

```diff
@@ -7,5 +7,9 @@ module SolidusAdmin
   class BaseComponent < ViewComponent::Base
     include ViewComponent::InlineTemplate
     include SolidusAdmin::ContainerHelper
+
+    def identifier
+      @identifier ||= module_parent.to_s.underscore.dasherize.gsub(%r{/}, '--')
```

Comment:
Two considerations:

- Should we remove the `solidus_admin` prefix?
- Following what you did in #5134, why not skip dasherizing and use `/` instead of `--` for the identifier?

The result of both is that the `SolidusAdmin::MainNav::Item::Component` component could be referenced with `main_nav/item`.

Reply:
We need to check if using `/` is valid for stimulus controller names, in that case I think it would be much clearer, although would create a discrepancy when controller names are used in data-attributes:

```html
<div data-controller="foo/bar">
  <span data-foo-bar-target="baz"></span>
</div>
```

As I don't think we can do:

```html
<div data-controller="foo/bar">
  <span data-foo/bar-target="baz"></span>
</div>
```

Reply:
> As I don't think we can do:
> 
> ```
> <div data-controller="foo/bar">
>   <span data-foo/bar-target="baz"></span>
> </div>
> ```

That's a good point. Maybe it's better to be consistent here?

Reply:
> Maybe it's better to be consistent here?

I'm not sure I understand what you're suggesting, can you expand on it?

Reply:
I was reconsidering if it's worth changing how identifiers look like if we can't use the same format everywhere (the data attribute you suggested").

Reply:
Oh I see, now it's clear 👍 

I think we should not conflate the registry keys and the internal component identifier, which should be independent and for stimulus is tied to the path of the javascript file. 

Maybe we can rename the method and call it `js_id` or `stimulus_id` (or `js_identifier` or `javascript_identifier` etc.), but since it's likely to be used often I would take shortness in consideration.

Thoughts on this?

[Norm Macdonald — ID (A Strange Abbreviation) [Izp52QUeSzQ].webm](https://github.com/solidusio/solidus/assets/1051/e97737e5-8397-4501-a5be-71b338e0d45d)

Reply:
> I think we should not conflate the registry keys and the internal component identifier, which should be independent and for stimulus is tied to the path of the javascript file.

I completely agree. Same for the `components` helper.

> Maybe we can rename the method and call it js_id or stimulus_id (or js_identifier or javascript_identifier etc.), but since it's likely to be used often I would take shortness in consideration.

Maybe `stimulus_id` makes everything clearer? No strong opinion, though.

Reply:
👍 Renamed to `stimulus_id`